src/map/mail.cpp

```diff
@@ -106,8 +109,28 @@ bool mail_removezeny( map_session_data *sd, bool flag ){
 	if( sd->mail.zeny > 0 ){
 		//Zeny send
 		if( flag ){
+			int64 fee;
+
+			if( util::safe_multiplication( static_cast<decltype(fee)>( sd->mail.zeny ), static_cast<decltype(fee)>( battle_config.mail_zeny_fee ), fee ) ){
+				return false;
+			}
+
+			if( fee < 0 ){
+				return false;
+			}
+
+			fee /= 100;
+
+			if( fee > MAX_ZENY ){
+				return false;
+			}
```

Comment:
```suggestion
			int64 zeny = sd->mail.zeny;

			if( battle_config.mail_zeny_fee > 0 ){
				int64 fee;

				if( util::safe_multiplication( zeny, static_cast<decltype(fee)>( battle_config.mail_zeny_fee ), fee ) ){
					return false;
				}

				if( fee < 0 ){
					return false;
				}

				fee /= 100;

				if( fee > MAX_ZENY ){
					return false;
				}

				if( util::safe_addition( zeny, fee, zeny ) ){
					return false;
				}

				if( zeny > MAX_ZENY ){
					return false;
				}
			}
```
Cleaned it up a little further. Also now summing up zeny+fee.